CMakeLists.txt

```diff
@@ -238,6 +265,7 @@ endif()
 target_link_libraries(dosbox PRIVATE
   $<TARGET_NAME_IF_EXISTS:SDL2::SDL2main>
   $<IF:$<TARGET_EXISTS:SDL2::SDL2>,SDL2::SDL2,SDL2::SDL2-static>
+  $<IF:$<BOOL:${SUPPORT_XINPUT2}>,${X11_Xinput_LIB},>
```

Comment:
Yeah, so no need for `SUPPORT_XINPUT2` to be defined here. You can do `target_link_libraries` in `src/libs/manymouse/CMakeLists.txt`, many other submodules in the project do it similarly. It's better to do these in the individual subfolder/submodules rather than in the global CMakeLists.txt.

Reply:
But how to I propagate the `C_MANYMOUSE` value to `src/hardware/input/` then?

Reply:
Yeah, I wasn't aware of that 😄 It's best to keep it this way then if `C_MANYMOUSE` is used in two different branches of the source tree.

You could use forced cache variables as global variables (see [here](https://cliutils.gitlab.io/modern-cmake/chapters/basics/variables.html#cache-variables)), but let's leave setting all `C_*` options in the top level CMakeLists.txt then.

Could you at least do the `target_link_libraries ... $<IF:$<BOOL:${SUPPORT_XINPUT2}>,${X11_Xinput_LIB},>` stuff in `src/libs/manymouse/CMakeLists.txt`? That would be better and could be done without complications, I think.

Reply:
Done, target link library moved.